include/rcutils/error_handling.h

```diff
@@ -314,6 +314,93 @@ RCUTILS_PUBLIC
 void
 rcutils_reset_error(void);
 
+// Helper macro for nested macro expansion
+#define _RCUTILS_EXPAND(x) x
+
+/// Set the error message, as well as append the current file, line number, and previous error.
+/**
+ * If there is no previous error, has same behavior as RCUTILS_SET_ERROR_MSG.
+ * \param[in] msg The error message to be set.
+ */
+#define RCUTILS_SET_ERROR_MSG_AND_APPEND_PREV_ERROR(msg) \
+  do { \
+    if (rcutils_error_is_set()) { \
```

Comment:
I would not add this as part of the macro. I think it encourages code that doesn't know if the called function is going to set the error. Instead I think it's better to assert one way or the other. 

The reason being that just recently I was debugging why a called function was failing and it returned error not set when I appended it. That told me which of the error checking statements was the cause and let me fix that. I think pretty much every error check that returns an error code should set the error message.

Reply:
To prevent raising an error or aborting in the case where the error isn't set, how about I print this instead?
```
      RCUTILS_SAFE_FWRITE_TO_STDERR("No previous error set\n"); \
```

Reply:
I don't think it errors or aborts. I think it just returns a string that says error not set.

Reply:
https://github.com/ros2/rcutils/blob/5bd1230eb10595cec30168667adf03f48879b96c/include/rcutils/error_handling.h#L305

Reply:
Ah, I misunderstood what you meant, I'll make the appropriate change